### ISSUE TITLE
False positive for `consider-using-f-string` when using modulo operator on a string

### ISSUE THREAD

**Issue by bgurney-rh** [PyCQA/pylint#6689]
### Bug description

```python
#!/usr/bin/python3

"""
Test unittest 1.
"""

import unittest

# pylint: disable=pointless-statement
class ModuloFailTestCase(unittest.TestCase):
    """Test modulo operator failure"""

    def test_int_str_modulo_fail(self):
        """Test int and str modulo operator failure"""
        with self.assertRaises(TypeError):
            1 % "str"
```

### Command used

```shell
pylint unittestprog1.py
```

### Pylint output

```shell
************* Module unittestprog1
unittestprog1.py:16:16: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
```

### Expected behavior

Do not flag a modulo (`%`) operation on a string with the `consider-using-f-string` lint.

(I discovered this issue while working on converting the justbytes module [ https://github.com/mulkieran/justbytes ] to formatted strings:)

```python
class ModTestCase(unittest.TestCase):
    """Test mod."""

    def test_exceptions(self):
        """Test that exceptions are thrown."""
        # pylint: disable=expression-not-assigned
        with self.assertRaises(RangeNonsensicalBinOpError):
            2048 % Range(12, B)
        with self.assertRaises(RangeNonsensicalBinOpError):
            Range(12) % "str"
        with self.assertRaises(RangeNonsensicalBinOpValueError):
            Range(12) % Range(0)
        with self.assertRaises(RangeNonsensicalBinOpValueError):
            Range(12) % 0
        with self.assertRaises(RangeNonsensicalBinOpError):
            Range(12) % Decimal("NaN")
```

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 12.0.1 20220308 (Red Hat 12.0.1-0)]

(This also occurs in the Fedora 36 package "pylint-2.13.7-1.fc36.noarch" .)
```

### OS / Environment

Fedora 36

**Comment by bgurney-rh** [comment]
Apologies; the description is supposed to be a reproducer program that can be executed via `python3 -m unittest -v unittestprog1.py` after saving the output to a file named "unittestprog1.py".
